include/nbl/asset/filters/kernels/CConvolutionWeightFunction.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_ASSET_C_CONVOLUTION_WEIGHT_FUNCTION_H_INCLUDED_
+#define _NBL_ASSET_C_CONVOLUTION_WEIGHT_FUNCTION_H_INCLUDED_
+
+#include "nbl/asset/filters/kernels/WeightFunctions.h"
+
+namespace nbl::asset
+{
+
+// If we allow the user to specify a derivative of CConvolutionWeightFunction1D then there will be the following problem:
+// There will be no way for us to evaluate the constituent functions (m_funcA, m_funcB) on an arbitrary derivative because
+// `derivative` is a class template member not a function template member (done to handle chain rule) of CWeightFunction1D.
+// So, we would have to create new `CWeightFunction1D`s entirely and use them (not just here, but in the various specializations
+// of this function). Not sure if its worth it, both in terms of performance and code complexity.
```

Comment:
explain what you mean?

Reply:
with convolution you have the derivative property
Derivative of convolution is the same as convolution where the first function is a derivative
https://discord.com/channels/593902898015109131/1043939034776473710/1064514311822774383

Reply:
> explain what you mean? I don't get the problem.

Ok I get it now

Reply:
Amend the comment and tell the user about the convolution property.

Basically that if they want "Nth order derivative of `Conv<CWeightFunction<A>,CWeightFunction<B>>`" then they should create "`Conv<CWeightFunction<A,i>,CWeightFunction<B,j>`" where `i<A::k_smoothness && j<B::k_smoothness && (i+j)==n`

Reply:
did it myself